lib/fastly_nsq/message_queue/consumer.rb

```diff
@@ -10,18 +10,21 @@ class Consumer
     def_delegator :connection, :size
     def_delegator :connection, :terminate
 
-    def initialize(topic:, channel:, ssl_context: nil)
-      @topic = topic
-      @channel = channel
+    def initialize(topic:, channel:, ssl_context: nil, &connector)
+      @topic       = topic
+      @channel     = channel
       @ssl_context = SSLContext.new(ssl_context)
+      @connector   = connector || DEFAULT_CONNECTOR
```

Comment:
I'm not sure I understand the objection to constructor-with-block, but since [we have a similar-but-different pattern in play elsewhere](https://github.com/fastly/fastly_nsq/blob/7d70eb82ec50200166b905d64db1924bf99f4a51/lib/fastly_nsq/message_queue/listener.rb#L6), it's certainly easy enough to follow along.